src/NuGet.Clients/PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -77,17 +79,9 @@ public bool CheckBoxesEnabled
 
         public bool IsSolution { get; set; }
 
-        private readonly ObservableCollection<object> _items = new ObservableCollection<object>(); 
-
-        public ObservableCollection<object> Items
-        {
-            get
-            {
-                Debug.Assert(Mvs.ThreadHelper.CheckAccess());
-                return _items;
-            }
-        }
+        private object _itemsLock = new object();
```

Comment:
We aren't updating it on UI thread so there should not be a scenario where it's blocking the UI. Do you still think we should go for WaitAsync?